sources/en.batcave/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+
+                        let title = Some(chapter
+                            .title
+                            .strip_prefix(&manga.title)
+                            .map(str::trim)
+                            .map(String::from)
+                            .unwrap_or_else(|| chapter.title));
+
+                        let date_uploaded = Some(
+                            chrono::NaiveDate::parse_from_str(&chapter.date, "%-d.%-m.%Y")
+                                .unwrap()
+                                .and_time(chrono::NaiveTime::MIN)
+                                .and_utc()
+                                .timestamp(),
+                        );
+
+                        Chapter {
+                            key: url.clone(),
+                            url: Some(url),
+                            title,
+                            date_uploaded,
+                            ..Default::default()
+                        }
+                    })
+                    .collect::<Vec<Chapter>>(),
+            );
+        }
+
+        Ok(manga)
+    }
+
+    fn get_page_list(&self, _manga: Manga, chapter: Chapter) -> Result<Vec<Page>> {
+        let html = Request::get(&chapter.key)?.html()?;
+
+        let pages = html
+            .select("script")
+            .map(|elements| {
+                elements
+                    .filter_map(|element| {
+                        let text = element.data()?;
+                        if !text.starts_with("window.__DATA__") {
+                            return None;
+                        }
+
+                        let page_list: PageList = serde_json::from_str(
+                            text.strip_prefix("window.__DATA__ = ")
+                                .and_then(|x| x.strip_suffix(";"))
+                                .unwrap_or_default(),
+                        )
+                        .unwrap();
+
+                        let pages = page_list
+                            .images
+                            .into_iter()
+                            .map(|page_url| Page {
+                                content: PageContent::url(page_url),
+                                ..Default::default()
+                            })
+                            .collect::<Vec<Page>>();
+
+                        Some(pages)
+                    })
+                    .flatten()
+                    .collect::<Vec<Page>>()
+            })
+            .unwrap_or_default();
+
+        Ok(pages)
+    }
+}
+
+impl Home for BatCave {
+    fn get_home(&self) -> Result<HomeLayout> {
+        let html = Request::get(BASE_URL)?.html()?;
+
+        let hot_releases_section = html
+            .select_first("main section.sect--hot")
+            .expect("No hot release section");
+
+        let title = hot_releases_section
+            .select_first("div")
+            .and_then(|x| x.text());
+
+        let hot_releases = hot_releases_section
+            .select("div>a")
```

Comment:
just for consistency
```suggestion
            .select("div > a")
```